llvm/lib/MC/MCParser/AsmParser.cpp

```diff
@@ -5095,18 +5095,16 @@ bool AsmParser::parseDirectiveComm(bool IsLocal) {
 
 /// parseDirectiveAbort
 ///  ::= .abort [... message ...]
-bool AsmParser::parseDirectiveAbort() {
-  // FIXME: Use loc from directive.
-  SMLoc Loc = getLexer().getLoc();
-
+bool AsmParser::parseDirectiveAbort(SMLoc DirectiveLoc) {
   StringRef Str = parseStringToEndOfStatement();
   if (parseEOL())
     return true;
 
   if (Str.empty())
-    return Error(Loc, ".abort detected. Assembly stopping.");
+    return Error(DirectiveLoc, ".abort detected. Assembly stopping.");
   else
-    return Error(Loc, ".abort '" + Str + "' detected. Assembly stopping.");
+    return Error(DirectiveLoc,
```

Comment:
While here, can you remove the `else`? https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return

The `return` below is unreachable, please remove it.

Reply:
Done, thx for noticing this